packages/core/src/index.ts

```diff
@@ -0,0 +1,46 @@
+import pluginBase from '@typescript-eslint/eslint-plugin';
+import * as parserBase from '@typescript-eslint/parser';
+import type { FlatConfig } from '@typescript-eslint/utils/ts-eslint';
+
+import allConfig from './configs/all';
+import baseConfig from './configs/base';
+import disableTypeCheckedConfig from './configs/disable-type-checked';
+import eslintRecommendedConfig from './configs/eslint-recommended';
+import recommendedConfig from './configs/recommended';
+import recommendedTypeCheckedConfig from './configs/recommended-type-checked';
+import strictConfig from './configs/strict';
+import strictTypeCheckedConfig from './configs/strict-type-checked';
+import stylisticConfig from './configs/stylistic';
+import stylisticTypeCheckedConfig from './configs/stylistic-type-checked';
+
+const parser: FlatConfig.Parser = {
+  meta: parserBase.meta,
+  parseForESLint: parserBase.parseForESLint,
+};
+
+const plugin: FlatConfig.Plugin = {
+  meta: pluginBase.meta,
+  rules: pluginBase.rules,
+};
+
+function flatConfig(...config: FlatConfig.ConfigArray): FlatConfig.ConfigArray {
+  return config;
+}
```

Comment:
Question:
What do we think about making this utility enforce a canonical name for plugins?

For example imagine like:
```ts
// eslint.config.js
import * as tseslint from '@typescript-eslint/core';

export default tseslint.config(
  {
    '@typescript-eslint': tseslint.plugin,
    // ...
  },
  {

  },
);
// or perhaps a more explicit object form like { plugins: { ... }, configs: [ ... ] } ?
// or perhaps a HOC form like tseslint.config({ ..plugins.. }, ...configs[]) ?
```

Then within the util we could rewrite all configs so that any references to the plugin use the defined namespace, and we rewrite it with pseudocode like:
```
for config in configs {
  const pair = config.plugins.entries().find(plugin.value)
  if pair && pair.key !== plugin.key {
    pair.key = plugin.key
    config.rules.keys().replace(`${pair.key}/`, `${plugin.key}/`)
  }
}
```

WDYT?

Reply:
I like the idea at first glance, but suspect it should be done in a follow-up issue so that the ESLint folks can provide input too.

Reply:
Based on the discussions we've had (https://github.com/eslint/eslint/discussions/17766) - this is fully intended behaviour and it doesn't look like they intend to change that?

Reply:
Oh I mean more on input for direction of the checking.

Plus splitting it out will make it easier to reference it when complaining about the name aliasing being allowed 😄